src/Lib/Element.php

```diff
@@ -14,185 +14,150 @@
 
     protected $elements = [];
 
+    protected $externalOptions = [];
+
+    protected $externalElements = [];
+
     /**
-     * Add an option to this element
-     * Options are generally string key/ mixed value pairs
-     *
-     * @param string $key
-     * @param mixed $value
+     * Return the class name of the element converted into snake_case
      *
-     * @return $this
+     * @return string
      */
-    final protected function addOption($key, $value)
+    public function getElementName()
     {
-        $this->options[$key] = $value;
-        return $this;
+        $classNameWithNamespace = get_class($this);
+        $classNameWithoutNamespace = substr($classNameWithNamespace, strrpos($classNameWithNamespace, '\\')+1);
+        preg_match_all('/[A-Z][a-z]+/', $classNameWithoutNamespace, $classNameParts);
```

Comment:
The only issue I can see is if ```strrpos($classNameWithNamespace, '\\')+1);``` returns a false positive because someone has named the class without a namespace. Which is very unlikely, and will break a lot of things if they do. It's up to you if you want to die hard here. Personally I would check the response of strpos() and preg_match_all(), but as you said if __all__ that goes wrong we have more issues to worry about.